### ISSUE TITLE
Add "auto-discovery" option for Ecto instrumentation

### ISSUE THREAD

**Issue by andrewhr** [open-telemetry/opentelemetry-erlang-contrib#47]
**TL;DR: provide option for users to setup Ecto telemetry with repo auto-discovery event, in addition to current explicit setup**

At current version (`1.0.0-rc.3`), Ecto setup is done explicit for each repo. Given Ecto provides repo init events, would be very useful to have some way to setup OpenTelemetry to listen those events and "auto-detect" repos. While the explicit approach we have today have many merits (doesn't rely on init order for example), with the auto-detect approach we have a bit more "drop-in" experience for Ecto. 

### Describe the solution you'd like

Some pseudo-code that can achieve this solution:

```elixir
  def setup(config \\ []) do
    :telemetry.attach({__MODULE__, :repo_init}, [:ecto, :repo, :init], &__MODULE__.handle_repo_init/4, config)
  end

  def handle_repo_init(_event_name, _measurements, metadata, _config) do
    event_prefix = metadata.opts[:telemetry_prefix] ++ [:query]
    OpentelemetryEcto.setup(event_prefix)
  end
```

### API and breaking changes

Assuming this feature is desirable, one open question that remains is how much this should be the "default" setup, or not. IMO, the best API is auto-first, but expose the explicit form. Something like:

```elixir
# auto discovery, uses setup_repo underneath
def setup(config \\ [])

# current 1.0.0-rc.3 setup implementation
def setup_repo(telemetry_prefix, config \\ [])
```

What I like most is `setup/1` is consistent with most cases so far, which instrument everything we may need automatically. The explicit version can have a more specific name like `setup_repo/2`, giving emphasis that works for a single repo at time.

The problem is **breaking changes**.

So, the backwards compatible change is to keep `setup/2` as-is, and including a second setup function that does auto-discovery.

### Conclusions

That's the feature request. If this feature fits OpenTelemetry vision, I'm open to work on some PR for that after guidance around breaking/non-breaking changes approach.
